OpenRA.Game/Support/AssemblyLoader.cs

```diff
@@ -0,0 +1,362 @@
+			foreach (var ext in NativeLibraryExtensions)
+			{
+				var candidate = Path.Combine(basePath, $"{prefix}{library.Name}{ext}");
+				if (File.Exists(candidate))
+				{
+					path = candidate;
+					return true;
+				}
+			}
+
+			// 2. Search in base path + app local (for portable deployments of netcoreapp)
+			var local = Path.Combine(basePath, library.AppLocalPath);
+			if (File.Exists(local))
+			{
+				path = local;
+				return true;
+			}
+
+			path = null;
+			return false;
+		}
+	}
+
+	public class ManagedLibrary
+	{
+		public AssemblyName Name { get; private set; }
+
+		public static ManagedLibrary CreateFromPackage(string assetPath)
+		{
+			return new ManagedLibrary
+			{
+				Name = new AssemblyName(Path.GetFileNameWithoutExtension(assetPath))
+			};
+		}
+	}
+
+	public class NativeLibrary
+	{
+		public string Name { get; private set; }
+
+		public string AppLocalPath { get; private set; }
+
+		public static NativeLibrary CreateFromPackage(string assetPath)
+		{
+			return new NativeLibrary
+			{
+				Name = Path.GetFileNameWithoutExtension(assetPath),
+				AppLocalPath = assetPath
+			};
+		}
+	}
+
+	public static class DependencyContextExtensions
+	{
+		public static AssemblyLoadContextBuilder TryAddDependencyContext(this AssemblyLoadContextBuilder builder, string depsFilePath, out Exception error)
+		{
+			error = null;
+			try
+			{
+				builder.AddDependencyContext(depsFilePath);
+			}
+			catch (Exception ex)
+			{
+				error = ex;
+			}
+
+			return builder;
+		}
+
+		public static AssemblyLoadContextBuilder AddDependencyContext(this AssemblyLoadContextBuilder builder, string depsFilePath)
+		{
+			var reader = new DependencyContextJsonReader();
+			using (var file = File.OpenRead(depsFilePath))
+			{
+				var deps = reader.Read(file);
+				builder.SetBaseDirectory(Path.GetDirectoryName(depsFilePath));
+				builder.AddDependencyContext(deps);
+			}
+
+			return builder;
+		}
+
+		static string GetFallbackRid()
+		{
+			string ridBase;
+
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				ridBase = "win10";
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+				ridBase = "linux";
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+				ridBase = "osx.10.12";
+			else
+				return "any";
+
+			switch (RuntimeInformation.OSArchitecture)
+			{
+				case Architecture.X86:
+					return ridBase + "-x86";
+				case Architecture.X64:
+					return ridBase + "-x64";
+				case Architecture.Arm:
+					return ridBase + "-arm";
+				case Architecture.Arm64:
+					return ridBase + "-arm64";
+			}
+
+			return ridBase;
+		}
+
+		public static AssemblyLoadContextBuilder AddDependencyContext(this AssemblyLoadContextBuilder builder, DependencyContext dependencyContext)
+		{
+			var ridGraph = dependencyContext.RuntimeGraph.Any()
+				? dependencyContext.RuntimeGraph
+				: DependencyContext.Default.RuntimeGraph;
+
+			var rid = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.GetRuntimeIdentifier();
```

Comment:
The PlatformAbstractions package [has been made obsolete in NET 5](https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/platformabstractions-package-removed), so if we decide to switch this PR back to NET 5 we should change this to use [RuntimeInformation.RuntimeIdentifier](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.runtimeidentifier?view=net-5.0).

Reply:
We cant do this, OpenRA.Game is a netstandard 2.1 lib.